### PR TITLE
Major Rewrite

### PR DESCRIPTION
**Pull request by Tech-TTGames**
Migrate config.json to a PostrgesSQL.

### CODE REVIEW

**Comment by kk5dire** [comment]
@Tech-TTGames How big are you planning on making v0.1

Also, 
Do you plan on using the channel description DB as opposed to a normal DB for channels still.
Of course, something would need to be done about things that require a per user DB, unless you would want to do something else in DMs for that? Or something that could be set individual to a user

**Comment by Tech-TTGames** [comment]
> @Tech-TTGames How big are you planning on making v0.1

Decently so but am not sure what I am going to add to it yet besides db.
> Also, Do you plan on using the channel description DB as opposed to a normal DB for channels still. Of course, something would need to be done about things that require a per user DB, unless you would want to do something else in DMs for that? Or something that could be set individual to a user
Eh. For v0.1 I'll probably use the full db.

**Comment by benfoster04** [comment]
> Migrate config.json to a PostrgesSQL.
A database is never recommended for a bot's configuration. I would heavily discourage its use, especially in this context where only minor improvements are meant to be made. This is not a fully-fledged ticket bot, it is simply an assistant.

JSON is an easy to use and easy to understand file format that everyone is happy with. It's not a worthwhile change. Plus, there are slash commands for everything! Don't complicate it, it'll make it inaccessible and unwelcoming to many

**Comment by Tech-TTGames** [comment]
> > Migrate config.json to a PostrgesSQL.
> 
> A database is never recommended for a bot's configuration. I would heavily discourage its use, especially in this context where only minor improvements are meant to be made. This is not a fully-fledged ticket bot, it is simply an assistant.
> 
> JSON is an easy to use and easy to understand file format that everyone is happy with. It's not a worthwhile change. Plus, there are slash commands for everything! Don't complicate it, it'll make it inaccessible and unwelcoming to many

The stuff that's important for setup will be in config.json still. Will just move all the more space-consuming stuff to db. The definition of the PR was a bit vauge.

**Comment by kk5dire** [comment]
> > Migrate config.json to a PostrgesSQL.
> 
> A database is never recommended for a bot's configuration. I would heavily discourage its use, especially in this context where only minor improvements are meant to be made. This is not a fully-fledged ticket bot, it is simply an assistant.
> 
> JSON is an easy to use and easy to understand file format that everyone is happy with. It's not a worthwhile change. Plus, there are slash commands for everything! Don't complicate it, it'll make it inaccessible and unwelcoming to many

There are benifts that can warrant having a database such as per user configs, allowing the program to be used in multiple guilds if ever wished to be, along with other data that can be stored there. I feel that for per channel data that would be fine to store in D, and other data (like user) if it can be. But there are some benifits imo to at the bare minimum *offering the ability to use a database*

**Comment by benfoster04** [comment]
Ok, by this point I believe a good plan would be to make the configuration modular. You'd obviously have to have an overruling json config, but enable the option to connect to a PostgresSQL, perhaps even options for MongoDB, or Firebase Firestore, and enable options for what segments of data are put where.

As my great uncle Ted always tells me, if you're going to overcomplicate it, don't half arse it

**Comment by Tech-TTGames** [comment]
> Ok, by this point I believe a good plan would be to make the configuration modular. You'd obviously have to have an overruling json config, but enable the option to connect to a PostgresSQL, perhaps even options for MongoDB, or Firebase Firestore, and enable options for what segments of data are put where.

Um… No? Or at least not by me. I'm going to make a db for more granular data storage and a minimized config.json. Moving basically everything that is not bot-wide config to the db. Cuz stuff can be per-guild, which does not make sense to keep in the json. Soooo… practically saying only the owner will stay in json. (And even that is really just an artifact of discord.py being stoopid with check and I don't have an idea how to get the bot object in those)

Basically, we're moving everything, so the bot can function in multiple guilds at once.

**Comment by kk5dire** [comment]
I +1 that TT, I assumed that was the case

**Comment by Tech-TTGames** [comment]
4e69fa9 has implemented #9

**Comment by Tech-TTGames** [comment]
We're moving out of draft! Will begin implementing features I can from issues!

**Comment by kk5dire** [comment]
GG!

**Comment by Tech-TTGames** [comment]
Boys it's review time!

**Review by kk5dire — APPROVED** [review]
LGTM

**Comment by Tech-TTGames** [comment]
Resolved issues with toolbox.

**Review by kk5dire — DISMISSED** [review]
Looks fine so far!

**Comment by kk5dire** [comment]
Appreciate the added clarifications to some comments.

**Comment by Tech-TTGames** [comment]
I AM NOW DECLARING:
OPERATION SHIPIT

**Comment by Tech-TTGames** [comment]
#52 
has been implemented.

**Comment by Tech-TTGames** [comment]
Resolved #52

**Review by kk5dire — DISMISSED** [review]
LGTM so far, as long as everything properly runs. :shipit:

**Comment by Tech-TTGames** [comment]
Merge at 21:50

File: tickets_plus/cogs/override.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+            ctx: The interaction context.
+        """
+        await ctx.response.send_message("Pulling latest changes...")
+        logging.info("Pulling latest changes...")
+        pull = await asyncio.create_subprocess_shell(
+            "git pull",
+            stdout=asyncio.subprocess.PIPE,
+            stderr=asyncio.subprocess.PIPE,
+            cwd=statvars.PROG_DIR,
+        )
```
Comment (benfoster04) [inline tickets_plus/cogs/override.py:129]: Because of the nature of this command as documented, I would highly advise a confirmation prompt be required as a secondary step before command execution continues.

**Review by benfoster04 — CHANGES_REQUESTED** [review]
Please see entry regarding confirmation prompts on dangerous commands. Otherwise LGTM and thank you for the tremendous work on the rewrite!

File: tickets_plus/cogs/override.py (hunk 1 above)
Comment (Tech-TTGames) [reply tickets_plus/cogs/override.py]: Done

**Review by benfoster04 — DISMISSED** [review]
Cleared for merge. Thanks again!

**Review by benfoster04 — APPROVED** [review]
